lightning/src/ln/functional_test_utils.rs

```diff
@@ -1098,6 +1115,26 @@ macro_rules! expect_payment_sent {
 			},
 			_ => panic!("Unexpected event"),
 		}
+		if $expect_paths {
+			for i in 1..events.len() {
+				match events[i] {
+					Event::PaymentPathSuccessful { .. } => {},
```

Comment:
Does it make sense to check the payment_id here matches the paymentsent event?

Reply:
Indeed. Also added for `payment_hash`, though that resulted in a failure in `ln::onion_route_tests::test_onion_failure`. I hacked in a workaround, but I'm wondering if there is a better way to make the tests less fragile / dependent on previous parts. Making them separate tests is probably the obvious way.